Zend/zend_execute.h

```diff
@@ -370,6 +370,33 @@ void zend_cleanup_unfinished_execution(zend_execute_data *execute_data, uint32_t
 		}                                                \
 	} while (0)
 
+#define ZEND_CLASS_HAS_TYPE_HINTS(ce) ((ce->ce_flags & ZEND_ACC_HAS_TYPE_HINTS) == ZEND_ACC_HAS_TYPE_HINTS)
+
+static zend_always_inline zend_bool _zend_object_has_type_hints(zval *object) {
+	do {
+		if (Z_TYPE_P(object) != IS_OBJECT) {
```

Comment:
In most cases this function is called when object already known to be IS_OBJECT, but performs useless check anyway. It;s better to inline ZEND_CLASS_HAS_TYPE_HINTS(Z_OBJCE_P(object)) in that places.